File: prototypes/image-repos/cuda-devel-redis/redis-utils/src/publisher.cpp
#include "publisher.h"

rds::Publisher::Publisher(std::string const &host, uint16_t port, std::string const &queue)
:RedisBase(host, port, queue){}

size_t rds::Publisher::publish(std::string const &item)
{
    return ctx -> rpush(_q_name, item);
}

File: prototypes/image-repos/cuda-redis-py/publisher.py
import time
import redis

class RedisPublisher:

    _q: redis.Redis
    #  Main queue
    _q_name: str  

    def __init__(self, 
        name: str, 
        host: str = "redis", 
        port: int = 6379, 
        db: int=0) -> None:
        self._q = redis.Redis(host=host, port=port, db=db)
        self._q_name = name
    
    def publish(self, count: int) -> None:
        for idx in range(1, count):
            print(f"Publishing: WorkItem-{idx}")
            self._q.rpush(self._q_name, f"WorkItem-{idx}")
            time.sleep(1)


if __name__ == "__main__":
    pass